node/node.go

```diff
@@ -0,0 +1,138 @@
+package node
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/url"
+	"os"
+	"slices"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nkeys"
+)
+
+type Node interface {
+	Validate() error
+	Start()
+}
+
+type nexNode struct {
+	nc *nats.Conn
+
+	logger                *slog.Logger
+	agentHandshakeTimeout int
+	resourceDirectory     string
+	internalNodeNATHost   string
```

Comment:
ultra mega tiny nit ... should be `internalNodeNATSHost` and internalNodeNATSPort`
you're missing the S in NATS

Reply:
though that internal host and port are things decided by the node... they shouldn't be configurable